examples/02_decoding/plot_haxby_searchlight_surface.py

```diff
@@ -39,64 +49,91 @@
 
 from sklearn import neighbors
 
-from nilearn import datasets, surface
-
-fsaverage = datasets.fetch_surf_fsaverage(mesh="fsaverage5")
-hemi = "left"
-
-radius = 5.0
-pial_mesh = fsaverage[f"pial_{hemi}"]
-X = surface.vol_to_surf(fmri_img, pial_mesh, radius=radius).T
+from nilearn import datasets
+from nilearn.experimental.surface import (
+    SurfaceImage,
+    load_fsaverage,
+)
 
-# %%
-# To define the :term:`BOLD` responses
-# to be included within each searchlight "sphere"
-# we define an adjacency matrix based on the inflated surface vertices such
-# that nearby surfaces are concatenated within the same searchlight.
+fsaverage = load_fsaverage()
+fmri_img_surf = SurfaceImage.from_volume(
+    mesh=fsaverage["pial"], volume_img=fmri_img, radius=5
+)
 
-infl_mesh = fsaverage[f"infl_{hemi}"]
-coords, _ = surface.load_surf_mesh(infl_mesh)
-radius = 3.0
-nn = neighbors.NearestNeighbors(radius=radius)
-adjacency = nn.fit(coords).radius_neighbors_graph(coords).tolil()
 
 # %%
 # Searchlight computation
 # -----------------------
+import numpy as np
 from sklearn.linear_model import RidgeClassifier
 from sklearn.model_selection import KFold
 from sklearn.pipeline import make_pipeline
 from sklearn.preprocessing import StandardScaler
 
 from nilearn.decoding.searchlight import search_light
 
-# Simple linear estimator preceded by a normalization step
-estimator = make_pipeline(StandardScaler(), RidgeClassifier(alpha=10.0))
+# initialize scores to be able to create a SurfaceImage from it later.
+scores = {
+    "left": np.zeros(fmri_img_surf.mesh.parts["left"].n_vertices),
+    "right": np.zeros(fmri_img_surf.mesh.parts["right"].n_vertices),
+}
+
+hemispheres_to_analyze = ["left", "right"]
```

Comment:
note that the previous example only ran the left hemisphere and now it runs both, so this run time of the example will be doubled

made it possible to only run it on one hemisphere

Reply:
I would run only one hemispheres to save resources.

Reply:
done